spot-vaults/contracts/BillBroker.sol

```diff
@@ -748,27 +738,33 @@ contract BillBroker is
         uint256 cutoff
     ) private pure returns (uint256) {
         if (arU <= cutoff) {
-            return _auc(fn1, arL, arU);
+            return _avgY(fn1, arL, arU);
         } else if (arL >= cutoff) {
-            return _auc(fn2, arL, arU);
+            return _avgY(fn2, arL, arU);
         } else {
-            return (_auc(fn1, arL, cutoff).mulDiv(cutoff - arL, arU - arL) +
-                _auc(fn2, cutoff, arU).mulDiv(arU - cutoff, arU - arL));
+            return (_avgY(fn1, arL, cutoff).mulDiv(cutoff - arL, arU - arL) +
+                _avgY(fn2, cutoff, arU).mulDiv(arU - cutoff, arU - arL));
         }
     }
 
-    /// @dev Given a linear function defined by points (x1,y1) (x2,y2),
-    ///      we compute the area under the curve between (xL, xU) assuming xL <= xU.
-    function _auc(Line memory fn, uint256 xL, uint256 xU) private pure returns (uint256) {
+    /// @dev We compute the average height of the line between {xL,xU}.
+    function _avgY(
+        Line memory fn,
+        uint256 xL,
+        uint256 xU
+    ) private pure returns (uint256) {
+        // if the line has a zero slope, return any y
+        if (fn.y1 == fn.y2) {
+            return fn.y2;
+        }
+
         // m = dlY/dlX
         // c = y2 - m . x2
-        // Integral m . x + c => m . x^2 / 2 + c
-        // Area between [xL, xU] => (m . (xU^2 - xL^2) / 2 + c . (xU - xL)) / (xU - xL)
-        //                       => m.(xL+xU)/2 + c
+        // Avg height => (yL + yU) / 2
+        //            => m . ( xL + xU ) / 2 + c
         int256 dlY = fn.y2.toInt256() - fn.y1.toInt256();
         int256 dlX = fn.x2.toInt256() - fn.x1.toInt256();
         int256 c = fn.y2.toInt256() - ((fn.x2.toInt256() * dlY) / dlX);
-        int256 area = ((xL + xU).toInt256() * dlY) / (2 * dlX) + c;
-        return area.abs();
+        return ((((xL + xU).toInt256() * dlY) / (2 * dlX)) + c).abs();
```

Comment:
Though the final avg height is always positive, intermediate values can be negative (the slope of the line, y intercept etc). So Im leaning toward keeping the math here as is (ie casting to signed int and casting back).